### PR TITLE
Configurator bugfix

### PR DESCRIPTION
**Pull request by zphang**
Fixes logic in computing max_steps in configurator:

**Testing**: Informal testing in notebooks, but no proper testing yet unfortunately. The configurator does some file-system reads to determine the number of examples in the task (from the task cache), so we'll need build some more advanced testing methods for that. Also this code is still pretty WIP / experimental.

### CODE REVIEW

File: jiant/proj/main/scripts/configurator.py [hunk 1]
```diff
@@ -117,11 +119,7 @@ def create_config(self):
             task_name_list_dict["train_val"] = self.parse_task_name_list(
                 self.train_val_task_name_list
             )
-        full_task_name_list = [
-            task_name
-            for task_name_list in task_name_list_dict.values()
-            for task_name in task_name_list
-        ]
+        full_task_name_list = py_datastructures.get_unique_list_in_order(task_name_list_dict.values())
```
Comment (jeswan) [inline jiant/proj/main/scripts/configurator.py:122]: `set(task_name_list_dict)` accomplishes the same thing. any motivation for the import?

Comment (zphang) [reply jiant/proj/main/scripts/configurator.py:122]: I'm not sure if set has guarantees on the resulting order.

Comment (jeswan) [reply jiant/proj/main/scripts/configurator.py:122]: Sounds good. Thanks for explaining!

**Comment by zphang** [comment]
Mind approving again?

**Pull request merged**